### PR TITLE
SQS: Add cross-account support

### PR DESCRIPTION
**Pull request by viren-nadkarni**
This PR makes SQS queues resolve based on the account ID and region in the queue URL.

Previously this was based on request context.

### CODE REVIEW

**Review by baermat — APPROVED** [review]
LGTM :+1:

File: localstack/services/sqs/provider.py [hunk 1]
```diff
@@ -1276,28 +1276,28 @@ def _create_message_attribute_hash(message_attributes) -> Optional[str]:
...
 
-def get_queue_name_from_url(queue_url: str) -> str:
-    return queue_url.rstrip("/").split("/")[-1]
-
-
-def resolve_queue_name(
+def resolve_queue_location(
     context: RequestContext, queue_name: Optional[str] = None, queue_url: Optional[str] = None
-) -> str:
+) -> Tuple[str, Optional[str], str]:
```
Comment (thrau) [inline localstack/services/sqs/provider.py:1281]: is the region still optional here?

**Review by thrau — APPROVED** [review]
thanks for adding this @viren-nadkarni :100: 

should we stage this for 2.1?

File: localstack/services/sqs/provider.py (hunk 1 above)
Comment (viren-nadkarni) [reply localstack/services/sqs/provider.py:1281]: yep, when neither endpoint nor domain endpoint strategy is used

File: tests/integration/test_sqs.py [hunk 2]
```diff
@@ -85,6 +89,43 @@ def test_get_queue_url_contains_request_host(self, sqs_create_queue, monkeypatch
...
+        # but this is not enforced in LocalStack
+        # - AddPermission
+        # - CreateQueue
+        # - DeleteQueue
+        # - ListQueues
+        # - ListQueueTags
+        # - RemovePermission
+        # - SetQueueAttributes
+        # - TagQueue
+        # - UntagQueue
```
[the remark this answers is not in the document]
Comment (viren-nadkarni) [reply tests/integration/test_sqs.py:127]: b0a8b377 :+1:

**Pull request merged**